drivers/char/broadcom/vc_sm/vmcs_sm.c

```diff
@@ -2056,6 +2110,138 @@ static int vc_sm_ioctl_unlock(struct SM_PRIV_DATA_T *private,
 	return ret;
 }
 
+/*
+ * Import a contiguous block of memory to be shared with VC.
+ */
+int vc_sm_ioctl_import_dmabuf(struct SM_PRIV_DATA_T *private,
+			      struct vmcs_sm_ioctl_import_dmabuf *ioparam,
+			      struct dma_buf *src_dma_buf)
+{
+	int ret = 0;
+	int status;
+	struct SM_RESOURCE_T *resource = NULL;
+	struct VC_SM_IMPORT import = { 0 };
+	struct VC_SM_IMPORT_RESULT result = { 0 };
+	struct dma_buf *dma_buf;
+	struct dma_buf_attachment *attach = NULL;
+	struct sg_table *sgt = NULL;
+
+	/* Setup our allocation parameters */
+	if (src_dma_buf) {
+		get_dma_buf(src_dma_buf);
+		dma_buf = src_dma_buf;
+	} else {
+		dma_buf = dma_buf_get(ioparam->dmabuf_fd);
+	}
+	if (IS_ERR(dma_buf))
+		return PTR_ERR(dma_buf);
+
+	attach = dma_buf_attach(dma_buf, &sm_state->pdev->dev);
+	if (IS_ERR(attach)) {
+		ret = PTR_ERR(attach);
+		goto error;
+	}
+
+	sgt = dma_buf_map_attachment(attach, DMA_BIDIRECTIONAL);
+	if (IS_ERR(sgt)) {
+		ret = PTR_ERR(sgt);
+		goto error;
+	}
+
+	/* Verify that the address block is contiguous */
+	if (sgt->nents != 1) {
+		ret = -ENOMEM;
+		goto error;
+	}
+
+	import.type = ((ioparam->cached == VMCS_SM_CACHE_VC) ||
+		       (ioparam->cached == VMCS_SM_CACHE_BOTH)) ?
+				VC_SM_ALLOC_CACHED : VC_SM_ALLOC_NON_CACHED;
+	import.addr = (uint32_t)sg_dma_address(sgt->sgl);
+	import.size = sg_dma_len(sgt->sgl);
+	import.allocator = current->tgid;
+
+	if (*ioparam->name) {
```

Comment:
Lose the braces.

Reply:
Checkpatch disagrees due to braces on the else clause.
CHECK: braces {} should be used on all arms of this statement
#396: FILE: drivers/char/broadcom/vc_sm/vmcs_sm.c:2164:
+	if (*ioparam->name)
[...]
+	else {
[...]

Reply:
But the else clause doesn't need them either.

Reply:
Ack. Think I'd looked at a different instance of almost identical syntax.